### ISSUE TITLE
Add support for Workload Identity on the GKE

### ISSUE THREAD

**Issue by mruoss** [googleapis/google-auth-library-php#303]
**Is your feature request related to a problem? Please describe.**
When running an application on the GKE I would like to be able to get credentials for a workload identity. The class `GCECredentials` already implements a great deal, however it only gets the credentials for the default service account. For workload identities, I would like to specify the google service account (GSA) the credentials are fetched for.

 **Describe the solution you'd like**
By refactoring the `GCECredentials` it is fairly easy to pass the service account name as parameter in the constructor. Or maybe adding a separate class like GKEWorkloadIdentityCredentials that inherits from GCECredentials would make sense.

**Comment by bshaffer** [comment]
@mruoss can you explain how you were able to get everything working?

**Comment by mruoss** [comment]
Thanks @bshaffer, I just figured out, this is actually really needed. Reopening therefore.

**Comment by mruoss** [comment]
I thought it was working but in fact it wasn't. I got a token, but for the wrong GSA.

**Comment by bshaffer** [comment]
@mruoss so essentially, the ask is for `GCECredentials` to have an option to request more than just the default service account from the metadata server?

```php
$gce = new GCECredentials(null, null, null, null, $serviceAccountIdentity);
$accessToken = $gce->fetchAuthToken();
```

Behind the scenes this would just change the metadata URL paths from `*/default/*` to `*/$serviceAccountIdentity/*`:

I don't see how we could add this to `ApplicationDefaultCredentials`, as no other credential type has the flexibility of only needing the service account name.

I've added #304 for this feature. Please take a look and see if this solution will work for you!

**Comment by mruoss** [comment]
Hi @bshaffer 

Yes, that is exactly the request. #304 seems to do the job.

**Comment by mruoss** [comment]
@bshaffer Can you roughly guess how long it's gonna take for this feature to be released?

**Comment by bshaffer** [comment]
@mruoss we just merged the PR, so it's available now via the `dev-master` packagist release. I'll push a new version by the end of week. Does that work for you?

**Comment by mruoss** [comment]
That sounds perfect, thank you very much!

**Comment by bshaffer** [comment]
@mruoss this is now released in [`v1.13.0`](https://github.com/googleapis/google-auth-library-php/releases/tag/v1.13.0)!
